### PR TITLE
basic functionality and some actions undo/redoable

### PR DESCRIPTION
**Pull request by magnificus**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -570,49 +571,47 @@ Viewport position (scale): ${viewportScreenX}, ${Math.round(
       const modKey = isMac ? e.metaKey : e.ctrlKey;
       // console.log(e.key);
       if (!isEventComingFromWithinTextInput(e)) {
+        e.preventDefault();
```
Comment (fakob) [inline src/App.tsx:574]: I had added `preventDefault()` on the case level is so it does not preventDefault if no case match. The believe was that I did not want to preventDefault if a shortcut is used by another element. Not sure what exactly that use case would be, but could you check if these use cases still work? 
* shortcuts like ctrl+a to select all in nodes like codeEditor and textEditor
* browser shortcuts

File: src/App.tsx [hunk 2]
```diff
@@ -570,49 +571,47 @@ Viewport position (scale): ${viewportScreenX}, ${Math.round(
...
               setIsGraphSearchOpen((prevState) => !prevState);
               break;
             case 'e':
-              e.preventDefault();
               setShowEdit((prevState) => !prevState);
               break;
+            case 'z':
+              ActionHandler.undo();
+              break;
+            case 'r':
```
Comment (fakob) [inline src/App.tsx:595]: I suggest to use modKey && e.shiftKey && 'z' for redo. This is the browser convention.
And modKey && 'r' is usually used for refresh. I would probably not touch that shortcut.

Comment (fakob) [reply src/App.tsx:595]: Ok, just saw that in Windows world the convention in the browser is mixed between
* ctrl+Y (most apps, including Office)
* ctrl+shift+Z (Miro)
I usually check what Miro is doing as it is similar and widespread both on windows and mac. Then we do not have to differentiate between platforms.

**Review by fakob — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (magnificus) [reply src/App.tsx:574]: hmm OK can check

File: src/App.tsx (hunk 2 above)
Comment (magnificus) [reply src/App.tsx:595]: OK

Comment (magnificus) [reply src/App.tsx:595]: switching to ctrl shift z

File: src/App.tsx (hunk 1 above)
Comment (fakob) [reply src/App.tsx]: The browser command reloading the tab stopped working now (ctrl + r). I suggest to revert this change and add preventDefault again for every single shortcut.

Comment (fakob) [reply src/App.tsx]: Copy pasting also stopped working.

Comment (magnificus) [reply src/App.tsx]: Alright thought I had but apparently I forgot

**Pull request merged**